CHANGELOG.md

```diff
@@ -5,6 +5,22 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/)
 and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Added
+- Support for `yarpc.code` annotation on Thrift files to allow service owners to
+specify a YARPC code for each one of their Thrift exceptions.
+- Included two new response headers for TChannel inbound calls: one for the application error 
+name and one for the application error code.
+- Set the newly introduced headers for the Thrift encoding in inbound calls.
+
+### Changed
+- Stats and logs now include the Thrift exception name and the related YARPC code,
+ if one is set. For inbound calls they are extracted from the Thrift file, 
+ for outbound calls they will be extracted from the newly introduced headers.
+- If a YARPC code is specified for the Thrift exception, YARPC will classify it between
+a client or server failure like it would classify a `Status`, based on the code.
+If the YARPC code is not specified for the Thrift exception, it will continue to assume
+a client failure.
```

Comment:
Let's trim this down, a bit. We only need to note changes that are observable to users, so we can drop the bit about headers. Also, since we're only adding functionality and not changing existing behaviour, let's put this all under the `### Added` header

```suggestion
### Added
- Using the `yarpc.code` annotation, services may specify an associated
`yarpcerrors.Code` for Thrift exceptions.
- Metrics and logs now include Thrift exception names and related YARPC code, if
  annotated. If a `yarpc.code` annotation is specified for a Thrift exception,
  metrics will classify it as a client or server failure, like a `yarpcerrors`
  error. If the YARPC code is not specified for the Thrift exception, it will
  continue to be assumed a client failure.
```

Reply:
Let's also make sure this CHANGELOG entry indicates that this is TChannel/Thrift only (at least until we get HTTP finished)